### ISSUE TITLE
Are OneToOne relationships working?

### ISSUE THREAD

**Issue by mxmeunier** [stanlemon/rest-bundle#42]
Do you handle @OneToOne relationships ?
Something like:

    {
        "fieldA": "...",
        ...
        "b": {
             "fieldB": "...",
             ...
         }
    }

doesn't seem to work for me.

POST method: http://localhost/web/web/app_dev.php/api/a.
Where "b" would be the mappedBy field to A)

For more details, I've writte an issue on [stackoverflow](http://stackoverflow.com/questions/31047583/doctrine2-onetoone-isnt-working?noredirect=1#comment50119335_31047583).

PS: in your Lemon\RestBundle\Object\Manager.php you always make on persist, I think this is why my thing doesn't work.

**Comment by stanlemon** [comment]
@mxmeunier Is what you're trying to do similar to this test? https://github.com/stanlemon/rest-bundle/blob/master/Tests/Controller/ResourceControllerTest.php#L392-L426  I want to make sure I understand the use case. Any chance I could get you to write a test that fails for this scenario?

**Comment by mxmeunier** [comment]
It is almost similar, but I can't persist the child before perform the POST action.

What I need is to be able to do somthing like this:

    $request = $this->makeRequest(
            'POST',
            '/person',
            json_encode(array(
                'name' => 'Maxime Meunier',
                'wife' => array(
                    'name' => 'Lucky Lady',
                )
            ))
        );

Without persisting the $wife element before (as you do with $mother element in your test case). In my application, I need to create at the same time one object with all his children and children of children with only one call to the API.

I can't, or I don't know how to, make a test because this call:

     $response = $this->controller->postAction($request, 'person');

will result an error with:

     An exception occurred while executing 'INSERT INTO Person (name, husband_id) VALUES (?, ?)' with params [\"Lucky Lady\", null]:\\\n\\\nSQLSTATE[HY000]: General error: 515 General SQL Server error: Check messages from the SQL Server [515] (severity 16) [(null)].

I think that my problem is more about my developpement requirements... because I need to do weird things. :-(

PS: At the moment I use a OneToMany relationship instead of OneToOne, with a unique index on all my fields... And it works but this isn't the "good way" to do it.

**Comment by stanlemon** [comment]
@mxmeunier Looking at your code there, I really think that's this test - isn't it?
https://github.com/stanlemon/rest-bundle/blob/master/Tests/Controller/ResourceControllerTest.php#L356-L390

The only difference being 'mother' instead of 'wife'.  The wife property is not defined in the class the tests are using for the fixture, maybe this is related to the issue you're hitting?

It might be easiest if you fork the repo and open a PR with the code you see failing, so we can work on it from there.

**Comment by mxmeunier** [comment]
Oh, sorry. I didn't looked at others test. The one I have wrote was an example of my problem, I did not tested it. It was just an example of what it could be.

Indeed this test seems to handle my problem. Maybe the fact that I'm using SQL Server instead of MongoDB (I have a MssqlBundle to work with Doctrine2) makes this not working for me.

Unfortunatly, I'm in an internship and I've no time to allow in a fork, I'm already late. Because @OneToMany with a unique index works fine, I'll keep this method even if this isn't a "good way" to use the association.

Thanks for your help.

PS: sorry if my english wasn't good. This isn't my birth language.
